README.md

```diff
@@ -1,31 +1,59 @@
 # esp-println
 
-Provides `print!`, `println!` and `dbg!` implementations for various Espressif devices.
-
-- Supports ESP32, ESP32-C2/C3/C6, ESP32-H2, ESP32-S2/S3, and ESP8266
-- Dependency free (not even depending on `esp-hal`, one optional dependency is `log`, another is `critical-section`)
-- Supports JTAG-Serial output where available
-- Supports RTT (lacking working RTT hosts besides _probe-rs_ for ESP32-C3 and ESP32-C6)
-- `no-op` features turns printing into a no-op
-
-## Usage
-
-In your `Cargo.toml`, under `[dependencies]`, add:
-
-```
-cargo add esp-println --features esp32
+A library that provides `print!`, `println!`, `dbg!` implementations and
+logging capabilities for Espressif devices.
+- Supports all Espressif devices.
+- Supports different communication methods:
+    - UART (Default)
+    - JTAG-Serial (Only available in ESP32-C3, ESP32-C6, ESP32-H2, ESP32-S3)
+    - [RTT]
+    - No-op: Turns printing into a no-op
+- Supports [`defmt`] backend
+
+# Usage
+
+```toml
+esp-println = { version = "0.7.0", features = ["esp32c2"] }
 ```
-
-Replace `esp32` with your target, if necessary.
+or `cargo add esp-println --features esp32c2`
+It's important to specify your target device as feature.
 
 Then in your program:
 
-```
+```rust
 use esp_println::println;
 ```
 
 You can now `println!("Hello world")` as usual.
 
+# Features
+
+- There is one feature for each supported target: `esp32`, `esp32c2`,
+  `esp32c3`, `esp32c6`, `esp32h2`, `esp32s2`, `esp32s3` and `esp8266`.
+   - One of these features must be enabled.
+   - Only one of these features can be enabled at a time.
+- There is one feature for each supported communication method: `uart`,
+  `jtag-serial` and `rtt` `no-op`.
+    - Only one of these features can be enabled at a time.
+- `log`: Enables logging using [`log` crate].
+- `colors` enable colored logging.
+   - Only effective when using the `log` feature.
+- `critical-section` enables critical sections.
+- There are two ways to use this library with [`defmt`]:
+   - `defmt`: This is intended to be used with [`espflash`], see `-L/--log-format` argument of `flash` or `monitor` subcommands of `espflash` and `cargo-espflash`
+     Uses [rzCOBS] encoding and adds framing.
+   - `defmt-raw`: Raw defmt output without additional framing. ⚠️ Be careful when using this feature: you must only write
```

Comment:
I guess we should also mention probe-rs here? Based on the discussion in https://github.com/esp-rs/esp-println/issues/65, I guess we say if you want defmt over rtt to 

- Use defmt-rtt with the defmt macros
- Use panic-probe instead of esp-backtrace

Maybe we should remove rtt support in esp-println :thinking:

Reply:
> Maybe we should remove rtt support in esp-println

mhhhh - IIRC I only added it because there were some problems with https://crates.io/crates/rtt-target and it was unmaintained back then. But that changed 🤔

Reply:
wouldn't mind see it go away

Reply:
Does `panic-probe` actually work with our chips? I seem to recall trying and was unable to get it building. Perhaps I'm just misremembering, though.

Reply:
Just tried again:

```text
error: `panic-probe` only supports Cortex-M targets (thumbvN-none-eabi[hf])
  --> /Users/jesse/.cargo/registry/src/index.crates.io-6f17d22bba15001f/panic-probe-0.3.1/src/lib.rs:24:1
   |
24 | compile_error!("`panic-probe` only supports Cortex-M targets (thumbvN-none-eabi[hf])");
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: could not compile `panic-probe` (lib) due to previous error
warning: build failed, waiting for other jobs to finish...
```

Haven't looked into the source yet, so not sure how much work is required to support different archs.

Reply:
I just mentioned probe-rs and this section, is there anything else we should address in this PR? I guess the discussion of dropping rtt support should take place in https://github.com/esp-rs/esp-println/issues/65

Reply:
Ugh, thats annoying, I looked at the code, only the "induce hard fault bit" is cortex-m specific. That could be put behind a feature. With that said, it seems defmt is getting less and less attention and they're offloading and deprecating many things. We can continue this in #65 like @SergioGasquez suggested.